### PR TITLE
add vuex state management support - not working

### PR DESCRIPTION
**Pull request by lucbic**
I can't make it work properly. The api for creating new todos seems to be working just fine. I just can't make to app to display all todos using  `this.$store.state.todos;`

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -15,6 +15,7 @@
   },
   "dependencies": {
     "axios": "^0.17.1",
+    "babel-polyfill": "^6.26.0",
```
Comment (badosu) [inline client/package.json]: You don't need this dependency

File: client/plugins/logger.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
+// Credits: borrowed code from fcomb/redux-logger
```
Comment (badosu) [inline client/plugins/logger.js]: You can remove this file

File: client/src/store/index.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import Vue from 'vue';
+import Vuex from 'vuex';
+import { state, mutations } from './mutations';
+import plugins from './plugins';
```
Comment (badosu) [inline client/src/store/index.js]: You can remove this line

File: client/src/store/index.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+Vue.use(Vuex);
+
+export default new Vuex.Store({
+  state,
+  mutations,
+  plugins,
```
Comment (badosu) [inline client/src/store/index.js]: And this line

File: client/src/store/mutations.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+export const STORAGE_KEY = 'todos-vuejs';
+
+// for testing
+if (navigator.userAgent.indexOf('PhantomJS') > -1) {
+  window.localStorage.clear();
+}
+
+export const state = {
+  todos:
```
Comment (badosu) [inline client/src/store/mutations.js]: This is the default state, you can make it equal to `[]`, i.e. by default all todos are empty before being retrieved.

File: client/src/store/plugins.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import { STORAGE_KEY } from './mutations';
```
Comment (badosu) [inline client/src/store/plugins.js]: You can remove this file

File: client/package.json (hunk 1 above)
Comment (lucbic) [reply client/package.json]: ok

File: client/plugins/logger.js (hunk 2 above)
Comment (lucbic) [reply client/plugins/logger.js]: ok. removed the whole `plugins` folder.

File: client/src/store/index.js (hunk 3 above)
Comment (lucbic) [reply client/src/store/index.js]: ok

File: client/src/store/index.js (hunk 4 above)
Comment (lucbic) [reply client/src/store/index.js]: ok

File: client/src/store/plugins.js (hunk 6 above)
Comment (lucbic) [reply client/src/store/plugins.js]: ok

File: client/src/store/mutations.js (hunk 5 above)
Comment (lucbic) [reply client/src/store/mutations.js]: ok

File: client/src/App.vue [hunk 7]
```diff
@@ -1,8 +1,10 @@
 <template>
   <div>
     <todo-list v-bind:todos="todos"></todo-list>
```
Comment (badosu) [inline client/src/App.vue]: This bind is unnecessary, you can use the store todos inside the component

File: client/src/App.vue [hunk 8]
```diff
@@ -19,22 +21,13 @@ export default {
     todos() {
```
Comment (badosu) [inline client/src/App.vue]: This is unnecessary as only the `TodoList` component is using it, move it there

**Comment by badosu** [comment]
LGTM 👍

**Pull request merged**